climbARUnity/Assets/Climbing Hold/ClimbingHold.cs

```diff
@@ -1,45 +1,83 @@
 ﻿using UnityEngine;
 using System.Collections;
+using System;
 
 public class ClimbingHold : MonoBehaviour
 {
-    //void OnTriggerEnter2D(Collider2D col)
-    //{
-    //    Debug.Log("There was a collision!");
-    //    gameObject.GetComponent<LineRenderer>().SetColors(UnityEngine.Color.green, UnityEngine.Color.green);
-    //}
-
     private int enterCount;
+    private System.DateTime lastCountedCollision;
+    private int smoothing = 750;
+
+    public bool smoothingEnabled;
 
-    void OnStart()
+    void Start()
     {
         enterCount = 0;
+        smoothingEnabled = true;
+        lastCountedCollision = System.DateTime.UtcNow;
+    }
+
+    public bool ShouldRegisterHoldGrabbed(Collider2D col)
+    {
+        if (col != null && col.gameObject.tag == "Hold")
+        {
+            return false;
+        }
+        else if (smoothingEnabled)
+        {
+            enterCount++;
+            System.DateTime currentTime = System.DateTime.UtcNow;
+            TimeSpan diff = currentTime - lastCountedCollision;
+
+            if (enterCount > 1 || diff.TotalMilliseconds < smoothing)
+            {
+                return false;
+            }
+
+            lastCountedCollision = System.DateTime.UtcNow;
+            return true;
+        }
+        else
+        {
+            return true;
+        }
     }
 
-    void OnUpdate()
+    public bool ShouldRegisterHoldReleased(Collider2D col)
     {
+        if (col != null && col.gameObject.tag == "Hold")
+        {
+            return false;
+        }
+        else if (smoothingEnabled)
+        {
+            enterCount--;
+            lastCountedCollision = System.DateTime.UtcNow;
 
+            return true;
+        }
+        else 
+        {
+            return true;
+        }
     }
 
-    void OnTriggerExit2D(Collider2D col)
+    public void OnTriggerExit2D(Collider2D col)
     {
-        enterCount--;
-        if (enterCount == 0)
+        if (ShouldRegisterHoldReleased(col))
         {
             gameObject.GetComponent<LineRenderer>()
-                .startColor = UnityEngine.Color.red;
+            .startColor = UnityEngine.Color.red;
```

Comment:
ohh wait actually @db2882 sorry, it seems like no matter what since the developer is override OnTriggerEnter/Exit they need to specify the line renderer color (based on what i see in MenuHold)? this seems a little annoying since it's boilerplate but not sure if there's anything great we can do about it